core/client/batch_client.go

```diff
@@ -107,7 +107,7 @@ func (c *Client) BatchCreateMutation(ctx context.Context, users []*User) ([]*ent
 	go func() {
 		defer close(rChan)
 		var wg sync.WaitGroup
-		for w := 1; w < (runtime.NumCPU() - 1); w++ {
+		for w := 1; w < runtime.NumCPU(); w++ {
```

Comment:
This has exactly the same problem if `NumCPU() == 1`, doesn't it?

Reply:
It does. The previous logic was effectively `NumCPU - 2`.  It's now `NumCPU -1`.
Would you recommend straight `NumCPU` or a fancy function to determine the parallelism?

Reply:
`w := 0` should fix the general problem, and then maybe you could do `w < max(1, runtime.NumCPU()-2)` if you wanted to be fancy -- although consider how intensive the `allocate` vs `work` functions are in the different use cases here - if the cost of doing the work is vastly greater than the `allocate` (e.g. signature/mutation verification), then it's probably better to just have a worker goroutine per core and let the scheduler steal slivers of time for the `allocate` bit.

Reply:
`allocate` work should be super minimal, happy to allow the scheduler to handle it. I was more worried about starving the entire node of resources -- the node might be busy doing other things too.

Reply:
OK, we'll use all the cpus :-)